src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Element<T> top;
+    private int size = 0;
+
+    private static class Element<T> {
+        private T value;
+        private Element<T> previous;
 
+        public Element(T value, Element<T> previous) {
+            this.value = value;
+            this.previous = previous;
+        }
+    }
+
+    public void push(T value) {
+        if (size == 0) {
+            top = new Element<>(value, null);
+        } else {
+            Element<T> notTheTopAnyMore = top;
+            top = new Element<>(value, notTheTopAnyMore);
+        }
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
```

Comment:
In the 'peek' method, to make your code cleaner, you can avoid using 'else' after 'throw'. When the 'throw' statement is executed, the control is immediately transferred out of the method. Therefore, there is no need to use 'else'.